src/main/oAuthClient.ts

```diff
@@ -276,6 +286,54 @@ export default class OAuthClient {
     })
   }
 
+  private loginWithIdToken(provider: string, idToken: string, nonce: string, opts: AuthOptions = {}): Promise<void> {
+    const authParams = this.authParams({
+      ...opts,
+      useWebMessage: false
+    })
+
+    return this.loginWithRedirect({
+      ...authParams,
+      provider,
+      idToken,
+      nonce,
+    })
+  }
+
+  private googleOneTap(opts: AuthOptions = {}): void {
+    if (this.config?.googleClientId) {
+      const nonce = randomBase64String()
+      const binaryNonce = Buffer.from(nonce, 'utf-8')
+
+      window.crypto.subtle.digest('SHA-256', binaryNonce).then(hash => {
```

Comment:
Consider returning a `Promise` here, as done for PKCE generation.

Reply:
Actually, you this does return `Promise<void>`, so the error path should return a failed promise.